src/charm.py

```diff
@@ -85,44 +81,49 @@ def _on_controller_relation_changed(self, event):
         `juju dashboard` command.
         """
         self._stored.controllerData["controller-url"] = event.relation.data[event.app]["controller-url"]
-        self._stored.controllerData["model-url-template"] = event.relation.data[event.app]["model-url-template"]
         self._stored.controllerData["identity-provider-url"] = event.relation.data[event.app].get("identity-provider-url", "")
         self._stored.controllerData["is-juju"] = event.relation.data[event.app]["is-juju"]
-        # Send the data to the controller for our endpoint
-        # XXX get the machines IP address using:
-        #   https://ops.readthedocs.io/en/latest/#ops.model.Network
-        #event.relation.data[event.app]["hostname"] = "0.0.0.0:8080"
-        # XXX render data into the html page
-        self.generate_and_save_config()
-        self.restart_nginx()
-
-    def generate_and_save_config(self):
+
+        self._configure()
+
+    def _configure(self):
         """
         Take the configuration values and render them to the index.html file
         for display on the static page.
         """
         data = self._stored.controllerData
+
+        controller_url = data.get("controller-url")
+        identity_provider_url = data.get("identity-provider-url")
+        is_juju = data.get("is-juju", False)
+
+        if not controller_url:
+            self.unit.status = BlockedStatus("Missing controller URL")
+            return
+
         env = Environment(loader=FileSystemLoader(os.getcwd()))
         env.filters['bool'] = bool
 
         config_template = env.get_template("src/config.js.template")
         config_template.stream(
-            controller_url=data.get("controller-url"),
             base_app_url="/",
-            identity_provider_url=data.get("identity-provider-url"),
-            is_juju=data.get("is-juju")
+            controller_api_endpoint="/api",
```

Comment:
Are we using `/api` vs `/controller/api` now?

Reply:
I thought it's just simpler api to revert to `controller/api`